src/mui/field/SelectField.js

```diff
@@ -0,0 +1,118 @@
+import React, { PropTypes } from 'react';
+import get from 'lodash.get';
+import pure from 'recompose/pure';
+import compose from 'recompose/compose';
+import translate from '../../i18n/translate';
+
+/**
+ * Display a value in an enumeration
+ *
+ * Pass possible options as an array of objects in the 'choices' attribute.
+ *
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <SelectField source="gender" choices={choices} />
+ *
+ * By default, the text is built by
+ * - finding a choice where the 'id' property equals the field value
+ * - using the 'name' property an the option text
+ *
+ * You can also customize the properties to use for the value and text,
+ * thanks to the 'optionValue' and 'optionText' attributes.
+ *
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <SelectField source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * `optionText` also accepts a function, so you can shape the option text at will:
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
+ * <SelectField source="author_id" choices={choices} optionText={optionRenderer} />
+ *
+ * `optionText` also accepts a React Element, that will be cloned and receive
+ * the related choice as the `record` prop. You can use Field components there.
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const FullNameField = ({ record }) => <Chip>{record.first_name} {record.last_name}</Chip>;
+ * <SelectField source="gender" choices={choices} optionText={<FullNameField />}/>
+ *
+ * The current choice is translated by default, so you can use translation identifiers as choices:
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'myroot.gender.male' },
+ *    { id: 'F', name: 'myroot.gender.female' },
+ * ];
+ *
+ * However, in some cases (e.g. inside a `<ReferenceField>`), you may not want
+ * the choice to be translated. In that case, set the `translateChoice` prop to false.
+ * @example
+ * <SelectField source="gender" choices={choices} translateChoice={false}/>
+ *
+ * **Tip**: <ReferenceField> sets `translateChoice` to false by default.
+ */
+export const SelectField = ({ source, record = {}, choices, elStyle, optionValue, optionText, translate, translateChoice = true }) => {
```

Comment:
Isn't the `translateChoice = true` redundant with the `defaultProps`? Same for `record`?